include/ada/urlpattern.h

```diff
@@ -0,0 +1,82 @@
+#ifndef ADA_URL_PATTERN_H
+#define ADA_URL_PATTERN_H
+
+#include "ada/common_defs.h"
+
+#include <unordered_map>
+#include <array>
+#include <string_view>
+#include <optional>
+
+namespace ada {
+
+ada_really_inline bool is_valid_name_code_point(const char32_t& c,
+                                                bool is_first) noexcept;
+
+struct urlpattern_options {
+  bool ignore_case = false;
+};
+
+struct component_result {
```

Comment:
`ada::component_result` seems a really vague naming, since it might be mistaken with `ada::url_components`